### DISCUSSION TITLE
Possible to support separate ClientId/secret per domain in same session.

### DISCUSSION THREAD

**Discussion by mrzaz** [acmesh-official/acme.sh#6119] [Ideas]
Hello,
I have a proposal if possible to support separate ClientId/secret per domain in same session ?
I have 2 wildcard domains that needs to be in same cert but from to separate Dynu accounts with separate OAuth API keys.

Account1 has one APIkey 1  *.domain1.tld
Account2 has a separate APIkey 2  *.domain2.tld

And need to generate it as:
./acme.sh --issue --dns dns_dynu --server letsencrypt -d *.domain1.tld -d *.domain2.tld
But as said, the ClientId/secret is different for domain1 compared to domain2.

I'm not even sure it is doable from acme.sh script point of view?

Best regards
Dan Lundqvist
Stockholm, Sweden

**Comment by neilpang** [comment]
Not supported -- dnsapi credentials are per provider account, not per domain.

Two options:
1. Issue two separate certs, one per domain.
2. DNS alias mode: CNAME `_acme-challenge.domain1.tld` and `_acme-challenge.domain2.tld` to a zone controlled by one account, then use `--challenge-alias`. See https://github.com/acmesh-official/acme.sh/wiki/DNS-alias-mode
